src/descriptors/mod.rs

```diff
@@ -259,26 +260,35 @@ impl LianaDescriptor {
             };
 
             // Unfortunately rust-miniscript satisfaction size estimation is inconsistent. For
-            // Taproot it considers the whole witness (including the control block size + the
-            // script size) but under P2WSH it does not consider the witscript! Therefore we
-            // manually add the size of the witscript, but only under P2WSH by the mean of the
-            // `explicit_script()` helper.
+            // Taproot it considers the whole witness (except the control block size + the
+            // script size), while under P2WSH it does not consider the witscript! Therefore we
+            // manually add the size of the witscript under P2WSH by means of the
+            // `explicit_script()` helper, which gives an error for Taproot, and for Taproot
+            // we add the sizes of the control block and script.
             let der_desc = self
                 .receive_desc
                 .0
                 .at_derivation_index(0)
                 .expect("unhardened index");
             let witscript_size = der_desc
                 .explicit_script()
-                .map(|s| varint_len(s.len()) + s.len())
-                .unwrap_or(0);
+                .map(|s| varint_len(s.len()) + s.len());
 
             // Finally, compute the satisfaction template for the primary path and get its size.
-            der_desc
-                .plan(&assets)
-                .expect("Always satisfiable")
-                .witness_size()
-                + witscript_size
+            let plan = der_desc.plan(&assets).expect("Always satisfiable");
+            plan.witness_size()
+                + witscript_size.unwrap_or_else(|_| {
+                    plan.witness_template()
+                        .iter()
+                        .map(|elem| match elem {
+                            // We need to calculate the size manually before calculating the varint length.
+                            // See https://docs.rs/miniscript/11.0.0/src/miniscript/util.rs.html#35-36.
+                            Placeholder::TapScript(s) => varint_len(s.len()),
+                            Placeholder::TapControlBlock(cb) => varint_len(cb.serialize().len()),
+                            _ => 0,
```

Comment:
So any stack item but the leaf script and control block account for the varint length? This is incredibly confusing, it'd be worth opening an issue there. (And if they fix it it would be nice to also fix https://github.com/rust-bitcoin/rust-miniscript/issues/701.)

Reply:
It appears so. I did some debugging to inspect all the elements and their sizes against an actual transaction and the sizes of these two placeholder elements didn't include the varint lengths.

I've created https://github.com/rust-bitcoin/rust-miniscript/issues/771 to track this.